File: ICARUS/Models/Icarus/Elements/ANCHOR.cs
using ICARUS.Models.Icarus.Elements.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ICARUS.Models.Icarus.Elements {

    /// <summary>
    /// Represents an html anchor A HREF 
    /// </summary>
    public class ANCHOR : EL {

        /// <summary>
        /// Constructs a generic HREF anchor
        /// </summary>
        /// <param name="model"></param>
        /// <param name="innerHtml">The text to be displayed within the ANCHOR tags</param>
        public ANCHOR(
            MODEL model, string innerHtml
        ) : base("A", model, innerHtml) {
            this.className = "ANCHOR";
        }
    }
}

File: Scripts/src/icarus/model/EL/a/anchor/ANCHOR.js
/** @module */
import A, { ATTRIBUTES, EL, MODEL, SPAN } from '../A.js';
import GLYPHICON, { DATA, MODELS } from '../../span/GLYPHICON.js';
/** A hyperlink / page anchor    
    @class
*/
export default class ANCHOR extends A {
	/** Constructs a generic Anchor
	    @param {EL} node Node
	    @param {AnchorModel} model Model
	*/
	constructor(node, model = MODELS.anchor()) {
		super(node, model);
		if (model.data.icon) {
			this.icon = new GLYPHICON(this, model);
		}
        this.label = new SPAN(this, MODELS.text(new ATTRIBUTES(), DATA.text(model.data.label)));
	}
}
export { A, ATTRIBUTES, DATA, EL, GLYPHICON, MODEL }